staff/kim-garcia/spotify-app/scripts/campo-de-tiro.js

```diff
@@ -0,0 +1,147 @@
+function SearchPanel() {
+    Component.call(this, 'form');
+
+    var input = document.createElement('input');
+    input.type = 'search';
+    input.placeholder = 'Input a text...';
+
+    var button = document.createElement('button');
+    button.type = 'submit';
+    button.innerHTML = 'Search';
+
+    this.element.appendChild(input);
+    this.element.appendChild(button);
+
+    var _callback;
+
+    this.element.addEventListener('submit', function (event) {
+        event.preventDefault();
+
+        var query = input.value;
+
+        if (query && _callback) _callback(query);
+    }.bind(this));
+
+    this.onSearch = function (callback) {
+        _callback = callback;
+    };
+}
+
+SearchPanel.prototype = Object.create(Component.prototype);
+SearchPanel.prototype.constructor = SearchPanel;
+
+var search = new SearchPanel();
+
+
+search.onSearch(function (query) {
+    
+    logic.searchBeers(query, function(beer){ 
+
+
+        results.updateResults(beer.map(function (result) {
+            return {
+                id: result.id,
+                text: result.name 
+            };
+        }));
+    
+        // detailContainer.clear();
+
+    });
+
+});
+
+function ResultsList() {
+    Component.call(this, 'ul');
+}
+
+ResultsList.prototype = Object.create(Component.prototype);
+ResultsList.prototype.constructor = ResultsList;
+
+ResultsList.prototype.updateResults = function (results) { // => { id, text }
+    this.element.innerHTML = '';
+
+    results.forEach(function (result) {
+        var li = document.createElement('li');
+        var a = document.createElement('a');
+
+        a.href = '#/' + result.id;
+        a.innerHTML = result.text;
+        a.onclick = function () {
+
+            if (this._callback) this._callback(result.id, result.text);
+
+
+        }.bind(this);
+
+        this.element.appendChild(li);
+
+        li.appendChild(a);
+    }, this);
+};
+
+ResultsList.prototype.onItemClick = function (callback) {
+    this._callback = callback;
+};
+
+var results = new ResultsList();
+
+results.onItemClick(function (id, text) {
```

Comment:
You should join the parametres to the function: 

`function (id, text) => function(id, text)`

delete the space between the word "function" and the "parentesis"